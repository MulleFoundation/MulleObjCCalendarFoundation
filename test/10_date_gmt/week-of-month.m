#ifdef __MULLE_OBJC__
# import <MulleObjCCalendarFoundation/MulleObjCCalendarFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif



int   main( void)
{
   NSCalendar         *calendar;
   NSDateComponents   *components;
   NSDate             *date;
   NSInteger          day;
   NSInteger          year;
   NSInteger          min;

   calendar   = [NSCalendar calendarWithIdentifier:NSGregorianCalendar];
   [calendar setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"GMT"]];
   [calendar setFirstWeekday:2];

   // incompatible with apple but shucks
   for( min = 1; min <= 7; min +=3)
   {
      [calendar setMinimumDaysInFirstWeek:min];
      printf( "--------%ld days in first week---------\n", min);
      for( year = 2017; year <= 2020; year++)
         for( day = -1; day < 15; day++)
         {
            components = [[NSDateComponents new] autorelease];
            [components setYear:year];
            [components setMonth:1];
            [components setDay:day];

            date       = [calendar dateFromComponents:components];
            components = [calendar components:~0
                                     fromDate:date];

            printf( "%ld.%ld.%ld %ld:%02ld:%02ld",
                  (long) [components day], (long) [components month], (long) [components year],
                  (long) [components hour], (long) [components minute], (long) [components second]);

            printf( " weekOfMonth: %ld\n",
                    (long) [components weekOfMonth]);
         }
   }
   return( 0);
}