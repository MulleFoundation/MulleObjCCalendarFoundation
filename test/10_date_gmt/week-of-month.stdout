--------1 days in first week---------
30.12.2016 0:00:00 weekOfMonth: 5
31.12.2016 0:00:00 weekOfMonth: 5
1.1.2017 0:00:00 weekOfMonth: 1
2.1.2017 0:00:00 weekOfMonth: 2
3.1.2017 0:00:00 weekOfMonth: 2
4.1.2017 0:00:00 weekOfMonth: 2
5.1.2017 0:00:00 weekOfMonth: 2
6.1.2017 0:00:00 weekOfMonth: 2
7.1.2017 0:00:00 weekOfMonth: 2
8.1.2017 0:00:00 weekOfMonth: 2
9.1.2017 0:00:00 weekOfMonth: 3
10.1.2017 0:00:00 weekOfMonth: 3
11.1.2017 0:00:00 weekOfMonth: 3
12.1.2017 0:00:00 weekOfMonth: 3
13.1.2017 0:00:00 weekOfMonth: 3
14.1.2017 0:00:00 weekOfMonth: 3
30.12.2017 0:00:00 weekOfMonth: 5
31.12.2017 0:00:00 weekOfMonth: 5
1.1.2018 0:00:00 weekOfMonth: 1
2.1.2018 0:00:00 weekOfMonth: 1
3.1.2018 0:00:00 weekOfMonth: 1
4.1.2018 0:00:00 weekOfMonth: 1
5.1.2018 0:00:00 weekOfMonth: 1
6.1.2018 0:00:00 weekOfMonth: 1
7.1.2018 0:00:00 weekOfMonth: 1
8.1.2018 0:00:00 weekOfMonth: 2
9.1.2018 0:00:00 weekOfMonth: 2
10.1.2018 0:00:00 weekOfMonth: 2
11.1.2018 0:00:00 weekOfMonth: 2
12.1.2018 0:00:00 weekOfMonth: 2
13.1.2018 0:00:00 weekOfMonth: 2
14.1.2018 0:00:00 weekOfMonth: 2
30.12.2018 0:00:00 weekOfMonth: 5
31.12.2018 0:00:00 weekOfMonth: 6
1.1.2019 0:00:00 weekOfMonth: 1
2.1.2019 0:00:00 weekOfMonth: 1
3.1.2019 0:00:00 weekOfMonth: 1
4.1.2019 0:00:00 weekOfMonth: 1
5.1.2019 0:00:00 weekOfMonth: 1
6.1.2019 0:00:00 weekOfMonth: 1
7.1.2019 0:00:00 weekOfMonth: 2
8.1.2019 0:00:00 weekOfMonth: 2
9.1.2019 0:00:00 weekOfMonth: 2
10.1.2019 0:00:00 weekOfMonth: 2
11.1.2019 0:00:00 weekOfMonth: 2
12.1.2019 0:00:00 weekOfMonth: 2
13.1.2019 0:00:00 weekOfMonth: 2
14.1.2019 0:00:00 weekOfMonth: 3
30.12.2019 0:00:00 weekOfMonth: 6
31.12.2019 0:00:00 weekOfMonth: 6
1.1.2020 0:00:00 weekOfMonth: 1
2.1.2020 0:00:00 weekOfMonth: 1
3.1.2020 0:00:00 weekOfMonth: 1
4.1.2020 0:00:00 weekOfMonth: 1
5.1.2020 0:00:00 weekOfMonth: 1
6.1.2020 0:00:00 weekOfMonth: 2
7.1.2020 0:00:00 weekOfMonth: 2
8.1.2020 0:00:00 weekOfMonth: 2
9.1.2020 0:00:00 weekOfMonth: 2
10.1.2020 0:00:00 weekOfMonth: 2
11.1.2020 0:00:00 weekOfMonth: 2
12.1.2020 0:00:00 weekOfMonth: 2
13.1.2020 0:00:00 weekOfMonth: 3
14.1.2020 0:00:00 weekOfMonth: 3
--------4 days in first week---------
30.12.2016 0:00:00 weekOfMonth: 5
31.12.2016 0:00:00 weekOfMonth: 5
1.1.2017 0:00:00 weekOfMonth: 1
2.1.2017 0:00:00 weekOfMonth: 2
3.1.2017 0:00:00 weekOfMonth: 2
4.1.2017 0:00:00 weekOfMonth: 2
5.1.2017 0:00:00 weekOfMonth: 2
6.1.2017 0:00:00 weekOfMonth: 2
7.1.2017 0:00:00 weekOfMonth: 2
8.1.2017 0:00:00 weekOfMonth: 2
9.1.2017 0:00:00 weekOfMonth: 3
10.1.2017 0:00:00 weekOfMonth: 3
11.1.2017 0:00:00 weekOfMonth: 3
12.1.2017 0:00:00 weekOfMonth: 3
13.1.2017 0:00:00 weekOfMonth: 3
14.1.2017 0:00:00 weekOfMonth: 3
30.12.2017 0:00:00 weekOfMonth: 5
31.12.2017 0:00:00 weekOfMonth: 5
1.1.2018 0:00:00 weekOfMonth: 1
2.1.2018 0:00:00 weekOfMonth: 1
3.1.2018 0:00:00 weekOfMonth: 1
4.1.2018 0:00:00 weekOfMonth: 1
5.1.2018 0:00:00 weekOfMonth: 1
6.1.2018 0:00:00 weekOfMonth: 1
7.1.2018 0:00:00 weekOfMonth: 1
8.1.2018 0:00:00 weekOfMonth: 2
9.1.2018 0:00:00 weekOfMonth: 2
10.1.2018 0:00:00 weekOfMonth: 2
11.1.2018 0:00:00 weekOfMonth: 2
12.1.2018 0:00:00 weekOfMonth: 2
13.1.2018 0:00:00 weekOfMonth: 2
14.1.2018 0:00:00 weekOfMonth: 2
30.12.2018 0:00:00 weekOfMonth: 5
31.12.2018 0:00:00 weekOfMonth: 6
1.1.2019 0:00:00 weekOfMonth: 1
2.1.2019 0:00:00 weekOfMonth: 1
3.1.2019 0:00:00 weekOfMonth: 1
4.1.2019 0:00:00 weekOfMonth: 1
5.1.2019 0:00:00 weekOfMonth: 1
6.1.2019 0:00:00 weekOfMonth: 1
7.1.2019 0:00:00 weekOfMonth: 2
8.1.2019 0:00:00 weekOfMonth: 2
9.1.2019 0:00:00 weekOfMonth: 2
10.1.2019 0:00:00 weekOfMonth: 2
11.1.2019 0:00:00 weekOfMonth: 2
12.1.2019 0:00:00 weekOfMonth: 2
13.1.2019 0:00:00 weekOfMonth: 2
14.1.2019 0:00:00 weekOfMonth: 3
30.12.2019 0:00:00 weekOfMonth: 6
31.12.2019 0:00:00 weekOfMonth: 6
1.1.2020 0:00:00 weekOfMonth: 1
2.1.2020 0:00:00 weekOfMonth: 1
3.1.2020 0:00:00 weekOfMonth: 1
4.1.2020 0:00:00 weekOfMonth: 1
5.1.2020 0:00:00 weekOfMonth: 1
6.1.2020 0:00:00 weekOfMonth: 2
7.1.2020 0:00:00 weekOfMonth: 2
8.1.2020 0:00:00 weekOfMonth: 2
9.1.2020 0:00:00 weekOfMonth: 2
10.1.2020 0:00:00 weekOfMonth: 2
11.1.2020 0:00:00 weekOfMonth: 2
12.1.2020 0:00:00 weekOfMonth: 2
13.1.2020 0:00:00 weekOfMonth: 3
14.1.2020 0:00:00 weekOfMonth: 3
--------7 days in first week---------
30.12.2016 0:00:00 weekOfMonth: 5
31.12.2016 0:00:00 weekOfMonth: 5
1.1.2017 0:00:00 weekOfMonth: 1
2.1.2017 0:00:00 weekOfMonth: 2
3.1.2017 0:00:00 weekOfMonth: 2
4.1.2017 0:00:00 weekOfMonth: 2
5.1.2017 0:00:00 weekOfMonth: 2
6.1.2017 0:00:00 weekOfMonth: 2
7.1.2017 0:00:00 weekOfMonth: 2
8.1.2017 0:00:00 weekOfMonth: 2
9.1.2017 0:00:00 weekOfMonth: 3
10.1.2017 0:00:00 weekOfMonth: 3
11.1.2017 0:00:00 weekOfMonth: 3
12.1.2017 0:00:00 weekOfMonth: 3
13.1.2017 0:00:00 weekOfMonth: 3
14.1.2017 0:00:00 weekOfMonth: 3
30.12.2017 0:00:00 weekOfMonth: 5
31.12.2017 0:00:00 weekOfMonth: 5
1.1.2018 0:00:00 weekOfMonth: 1
2.1.2018 0:00:00 weekOfMonth: 1
3.1.2018 0:00:00 weekOfMonth: 1
4.1.2018 0:00:00 weekOfMonth: 1
5.1.2018 0:00:00 weekOfMonth: 1
6.1.2018 0:00:00 weekOfMonth: 1
7.1.2018 0:00:00 weekOfMonth: 1
8.1.2018 0:00:00 weekOfMonth: 2
9.1.2018 0:00:00 weekOfMonth: 2
10.1.2018 0:00:00 weekOfMonth: 2
11.1.2018 0:00:00 weekOfMonth: 2
12.1.2018 0:00:00 weekOfMonth: 2
13.1.2018 0:00:00 weekOfMonth: 2
14.1.2018 0:00:00 weekOfMonth: 2
30.12.2018 0:00:00 weekOfMonth: 5
31.12.2018 0:00:00 weekOfMonth: 6
1.1.2019 0:00:00 weekOfMonth: 1
2.1.2019 0:00:00 weekOfMonth: 1
3.1.2019 0:00:00 weekOfMonth: 1
4.1.2019 0:00:00 weekOfMonth: 1
5.1.2019 0:00:00 weekOfMonth: 1
6.1.2019 0:00:00 weekOfMonth: 1
7.1.2019 0:00:00 weekOfMonth: 2
8.1.2019 0:00:00 weekOfMonth: 2
9.1.2019 0:00:00 weekOfMonth: 2
10.1.2019 0:00:00 weekOfMonth: 2
11.1.2019 0:00:00 weekOfMonth: 2
12.1.2019 0:00:00 weekOfMonth: 2
13.1.2019 0:00:00 weekOfMonth: 2
14.1.2019 0:00:00 weekOfMonth: 3
30.12.2019 0:00:00 weekOfMonth: 6
31.12.2019 0:00:00 weekOfMonth: 6
1.1.2020 0:00:00 weekOfMonth: 1
2.1.2020 0:00:00 weekOfMonth: 1
3.1.2020 0:00:00 weekOfMonth: 1
4.1.2020 0:00:00 weekOfMonth: 1
5.1.2020 0:00:00 weekOfMonth: 1
6.1.2020 0:00:00 weekOfMonth: 2
7.1.2020 0:00:00 weekOfMonth: 2
8.1.2020 0:00:00 weekOfMonth: 2
9.1.2020 0:00:00 weekOfMonth: 2
10.1.2020 0:00:00 weekOfMonth: 2
11.1.2020 0:00:00 weekOfMonth: 2
12.1.2020 0:00:00 weekOfMonth: 2
13.1.2020 0:00:00 weekOfMonth: 3
14.1.2020 0:00:00 weekOfMonth: 3
